### PR TITLE
nixos/sx: init

### PR DESCRIPTION
**Pull request by figsoda**
###### Motivation for this change

sx is a simple alternative to both xinit and startx for starting an Xorg server.

https://github.com/Earnestly/sx

this module is largely based on services.xserver.displayManager.startx

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Not an Xorg user, not sure why you are requesting my review.

**Comment by figsoda** [comment]
my bad, it was recommendded

File: nixos/modules/services/x11/display-managers/sx.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+let cfg = config.services.xserver.displayManager.sx;
+
+in {
+  options = {
+    services.xserver.displayManager.sx = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable the "sx" pseudo-display manager, which allows users
```
Comment (bobby285271) [inline nixos/modules/services/x11/display-managers/sx.nix:14]: Maybe `mkEnableOption`?

Comment (figsoda) [reply nixos/modules/services/x11/display-managers/sx.nix:14]: `mkEnableOption` specifies a short description, not sure if thats the right thing to do here

Comment (ryantm) [reply nixos/modules/services/x11/display-managers/sx.nix:14]: You could do something like:
```
enable = (mkEnableOption "'sx' pseudo-display manager") // { description = ''
...
''})
```

**Comment by figsoda** [comment]
resolved merge conflicts

**Comment by figsoda** [comment]
resolved merge conflicts

**Pull request merged**